contracts/src/v0.8/dev/VRFV2Wrapper.sol

```diff
@@ -0,0 +1,260 @@
+  // fees, so this should be set to the highest gas lane on the network.
+  bytes32 s_keyHash;
+
+  // s_maxNumWords is the max number of words that can be requested in a single wrapped VRF request.
+  uint8 s_maxNumWords;
+
+  mapping(uint256 => address) /* requestID */ /* callbackAddress */
+    public s_callbacks;
+
+  constructor(
+    address _link,
+    address _linkEthFeed,
+    address _coordinator
+  ) ConfirmedOwner(msg.sender) VRFConsumerBaseV2(_coordinator) {
+    LINK = LinkTokenInterface(_link);
+    LINK_ETH_FEED = AggregatorV3Interface(_linkEthFeed);
+    COORDINATOR = ExtendedVRFCoordinatorV2Interface(_coordinator);
+
+    // Create this wrapper's subscription and add itself as a consumer.
+    uint64 subId = ExtendedVRFCoordinatorV2Interface(_coordinator).createSubscription();
+    SUBSCRIPTION_ID = subId;
+    ExtendedVRFCoordinatorV2Interface(_coordinator).addConsumer(subId, address(this));
+  }
+
+  /**
+   * @notice setConfig configures VRFV2Wrapper.
+   *
+   * @dev Sets wrapper-specific configuration based on the given parameters, and fetches any needed
+   * @dev VRFCoordinatorV2 configuration from the coordinator.
+   *
+   * @param _wrapperGasOverhead reflects the gas overhead of the wrapper's fulfillRandomWords
+   *        function.
+   * @param _wrapperPremiumPercentage is the premium ratio in percentage for wrapper requests.
+   * @param _keyHash to use for requesting randomness.
+   */
+  function setConfig(
+    uint32 _wrapperGasOverhead,
+    uint8 _wrapperPremiumPercentage,
+    bytes32 _keyHash,
+    uint8 _maxNumWords
+  ) external onlyOwner {
+    s_wrapperGasOverhead = _wrapperGasOverhead;
+    s_wrapperPremiumPercentage = _wrapperPremiumPercentage;
+    s_keyHash = _keyHash;
+    s_maxNumWords = _maxNumWords;
+    s_configured = true;
+
+    // Get other configuration from coordinator
+    (, , s_stalenessSeconds, s_gasAfterPaymentCalculation) = COORDINATOR.getConfig();
+    s_fallbackWeiPerUnitLink = COORDINATOR.getFallbackWeiPerUnitLink();
+  }
+
+  /**
+   *
+   */
+  function getConfig()
+    external
+    view
+    returns (
+      int256 fallbackWeiPerUnitLink,
+      uint32 stalenessSeconds,
+      uint32 fulfillmentFlatFeeLinkPPM,
+      uint32 gasAfterPaymentCalculation,
+      uint32 wrapperGasOverhead,
+      uint8 wrapperPremiumPercentage,
+      bytes32 keyHash,
+      uint8 maxNumWords
+    )
+  {
+    return (
+      s_fallbackWeiPerUnitLink,
+      s_stalenessSeconds,
+      s_fulfillmentFlatFeeLinkPPM,
+      s_gasAfterPaymentCalculation,
+      s_wrapperPremiumPercentage,
+      s_wrapperPremiumPercentage,
+      s_keyHash,
+      s_maxNumWords
+    );
+  }
+
+  function calculatePrice(uint32 callbackGasLimit) public view override onlyConfiguredNotDisabled returns (uint256) {
+    return calculatePriceInternal(callbackGasLimit, tx.gasprice);
+  }
+
+  function estimatePrice(uint32 _callbackGasLimit, uint256 _requestGasPrice)
```

Comment:
Nit: Might be worth being super verbose here and calling the gas price parameter `_requestGasPriceWei` (since it's common that it gets specified in gwei instead).

Reply:
Done